### ISSUE TITLE
Almost no (estimated) training time reduction when increasing Micro Batch Size from 1 to 4

### ISSUE THREAD

**Issue by DavidFarago** [oobabooga/text-generation-webui#2125]
### Describe the bug

I am fine-tuning Vicuna on 50k conversations.

With Micro Batch Size 1, wandb shows 16GB memory usage, and the estimated runtime is 156 hours, 
With Micro Batch Size 4, wandb shows 46GB memory usage, but the estimated runtime only drops very little: to 142 hours.

I would have expected a runtime of about 40 hours (156/4). 

Are the estimates bad? In that case, is 156 hours too low or is 142 hours too high an estimate?

If the estimates are roughly correct, why do I not get a significant speedup when increasing the Micro Batch Size from 1 to 4 and keeping all other settings the same?

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Create (cannot hand mine out, sorry) 50k conversations as json lines, each with the following structure (following Vicuna 1.1 prompt):
```
{"conversation": "You are operating a robot that is helping a caller foo bar\n\nASSISTANT: Welcome\nUSER: Hi, this is Michael Fox. I would like to foo bar\nASSISTANT: Yes, I can help you foo bar\nUSER: I want an appointment at 6pm.\nASSISTANT: Which type of appointment?\nUSER: Type foo bar\nASSISTANT: I booked an appointment at 6pm of type foo bar"}
```
Use the following format:
```
{
    "conversation": "%conversation%"
}
```
Use the following settings: BS 128, Epochs 3, LR 3e-4 linear scheduler, rank 32, alpha 64, cutoff 2048.

### Screenshot

![bugReportMicroBatchSize](https://github.com/oobabooga/text-generation-webui/assets/1004626/72b38a1f-4794-4c58-8fcc-6fd155703f15)

### Logs

```shell
There is no error message, the error lies either in the estimated runtime or in the fine-tuning not being 4 times faster when increasing Micro Batch Size by a factor 4.
```

### System Info

```shell
Paperspace A4000: 45GiB CPU memory, 8 CPUs, Ampere A4000 GPU, 48GiB GPU Memory
```

**Comment by Ph0rk0z** [comment]
The estimates for me ended up being pretty accurate.

**Comment by DavidFarago** [comment]
> The estimates for me ended up being pretty accurate.
Thanks @Ph0rk0z, so I guess there really is no 4 time training speedup :( Do you have an idea why? What Micro Batch Size are you using?

**Comment by Ph0rk0z** [comment]
I just used batch size of 4 because it was default. No idea why, didn't even know it was supposed to provide a speedup.
